src/iccpd/src/scheduler.c

```diff
@@ -742,32 +831,66 @@ int scheduler_unregister_sock_read_event_callback(struct CSM* csm)
 void scheduler_session_disconnect_handler(struct CSM* csm)
 {
     struct System* sys = NULL;
+    struct If_info * cif = NULL;
 
     if ((sys = system_get_instance()) == NULL )
         return;
 
-    struct epoll_event event;
-
     if (csm == NULL)
         return;
 
+    ICCPD_LOG_NOTICE("ICCP_FSM", "scheduler session disconnect handler");
+
     session_conn_thread_lock(&csm->conn_mutex);
     scheduler_unregister_sock_read_event_callback(csm);
     if (csm->sock_fd > 0)
     {
-        event.data.fd = csm->sock_fd;
-        event.events = EPOLLIN;
-        epoll_ctl(sys->epoll_fd, EPOLL_CTL_DEL, csm->sock_fd, &event);
-
-        close(csm->sock_fd);
-        csm->sock_fd = -1;
+        scheduler_csm_socket_cleanup(csm, 1);
```

Comment:
with cli change from mclagsyncd, such as set_peer_link, will call scheduler_session_disconnect_handler at the end of function, at this time mclagsyncd connection will disconnect, why? mclagsyncd will blocked in accept all the time if no reconnect later, OR mclagsyncd will fall in reconnect every time if reconnect later since mclaglink class will create every time and sync domain config again!